### PR TITLE
fix: ensure ping back messages are called back and empty updates excluded

### PR DESCRIPTION
**Pull request by bdraco**
### Description of change

There was a check for empty device updates ealier in the function but we did not check again after unifi_dict_to_dict so we did a lot of useless callbacks

## Summary by CodeRabbit

- **New Features**
  - Introduced new functionality for handling empty dictionaries in WebSocket packet processing.

- **Bug Fixes**
  - Added conditional checks to prevent errors when handling empty data.

- **Tests**
  - Added tests for the `ModelType` enum's `from_string` method to verify correct string-to-enum conversions.
  - Updated WebSocket emission tests to include new conditions and assertions.

### CODE REVIEW

**Comment by bdraco** [comment]
Needs to be tested with a doorbell to make sure ring still works

**Comment by darkrain-nl** [comment]
> DRAFT until this can be tested with doorbells (which I do not have)

Is there a possibility for me to help with this testing? I assume I need to be on a newer HA first... what else is needed as this is not a PR in HA so the script provided earlier probably won't work?

I don't have time today, but maybe tomorrow or else next week I could try...

**Comment by bdraco** [comment]
If you could manually patch the the installed library with this change and test they would be great

No worries if you can't. I'm going to visit family next week and will be able to test it than (assuming it won't be too annoying to ring their doorbell a few times)

**Comment by bdraco** [comment]
I got an old test doorbell that I hooked up via usb-c.
This doesn't break it because event pingbacks are already broken!

**Comment by bdraco** [comment]
the broken pingbacks were not obvious because there is a full poll done every 20s which reset the doorbell ring status.

I'll adjust the integration to only do the full poll every 60s after I fix the pingbacks here since a full poll is quite heavy.

File: tests/test_api_ws.py [hunk 1]
```diff
@@ -449,6 +450,18 @@ async def test_ws_emit_ring_callback(
     mock_now.return_value = utc_now() + EVENT_PING_INTERVAL
     assert not obj.is_ringing
 
+    # The event message should be emitted
+    assert protect_client.emit_message.call_count == 1
+
+    await asyncio.sleep(0)
+    await asyncio.sleep(0)
+
```
[the remark this answers is not in the document]
Comment (bdraco) [reply tests/test_api_ws.py:458]: This is a bad suggestion. The test needs the event loop to run twice for it to work

**Pull request merged**